src/gnu/trove/array/AbstractOffheapArray.java

```diff
@@ -141,4 +149,60 @@ public Field run() {
             }
         });
     }
+
+    private static Object createCleaner(Object obj, Runnable cleanUpRunnable) {
+        if (createMethod == null) {
+            initializeCreateMethod();
+        }
+        try {
+            return createMethod.invoke(null, obj, cleanUpRunnable);
+        } catch (ReflectiveOperationException e) {
+            throw new IllegalStateException("Could not create a new cleaner.", e);
+        }
+    }
+
+    private static void initializeCreateMethod() {
```

Comment:
I think these methods need some synchronization. I don't remember off-hand all the options for making them thread-safe, but one option is that they could be synchronized and return the Method itself, vs. making the user of the method also read it separately. (i.e., this method starts with "if createMethod isn't null, return it".)

Reply:
Is there any concern here other than we may just initialize it twice? I don't think synchronization is needed for correctness, right, this is just an optimization for the case where we try to initialize it twice from different threads, right?

Reply:
There might be a special case around these reflection-related classes where it's not an issue, but in general I'd expect both read and write access to a shared static field to require some sort of synchronization (and I'm being paranoid accordingly).